use-cases.md

```diff
@@ -0,0 +1,184 @@
+  * Because the HTTP client may throw an exception even before returning a Future, we use a try/catch block to ensure that we finish the span in all circumstances.
+
+
+### Using Distributed Context Propagation
+
+The client and server examples above took care of propagating Trace Context on the wire, which includes any Trace Attributes.  The client may use the Trace Attributes to pass additional data to the server and any other downstream server it might call.
+
+```python
+
+    # client side
+    span.trace_context.set_attribute('auth-token', '.....')
+
+    # server side (one or more levels down from the client)
+    token = span.trace_context.get_attribute('auth-token')
+```
+
+### Logging Events
+
+We have already used `log_event_with_payload` in the client span use case. Events can be logged without a payload, and not just where the span is being created / finished. For example, the application may record a cache miss event in the middle of execution, as long as it can get access to the current span from request context:
+
+```python
+
+    span = get_current_span()
+    span.log_event('cache-miss') 
+```
+
+The tracer automatically records a timestamp of the event, in contrast with tags that apply to the entire span. In the future versions of the API it will be possible to associate an externally provided timestamp with the event, e.g. see [pull request #38](https://github.com/opentracing/opentracing-go/pull/38).
+
+### Recording Spans With External Timestamps
+
+There are scenarios when it is impractical to incorporate an OpenTracing compatible tracer into a service, for various reasons. For example, a user may have a log file of what's essentially span data coming from a black-box process (e.g. HAProxy). In order to get the data into an OpenTracing-compatible system, the API needs a way to record spans with extrenally captured timestamps. The current version of API does not have that capability: see [issue #20](https://github.com/opentracing/opentracing.github.io/issues/20).
+
+### Setting "Debug" Mode
+
+Most tracing systems apply sampling to minimize the amount of trace data sent to the system.  Sometimes developers want to have a way to ensure that a particular trace is going to be recorded (sampled) by the tracing system, e.g. by including a special parameter in the HTTP request, like `debug=true`. The OpenTracing API does not have any insight into sampling techniques used by the implementation, so there is no explicit API to force it. However, the implementations are advised to recognized the `debug` Trace Attribute and take measures to record the span. In order to pass this attribute to tracing systems that rely on pre-trace sampling, the following approach can be used:
+
+```python
+
+    if request.get('debug'):
+        trace_context = tracer.new_root_trace_context()
+        trace_context.set_attribute('debug', True)
```

Comment:
>
> DEBUG isn't intuitive for me either. I understand the SAMPLED isn't either.
>
> What about FORCE or FORCE_TRACING ?
> Because *debug* does imply something else.
>
Yuri did mention a use case I hadn't thought about, which was to highlight
visually "Debug" traces (since the state is persisted). Debug may not be
the right word, but maybe force doesn't imply what he meant?

At any rate, it seems we are choosing words that aren't defined in OT as an
alternative to words also not defined in OT :P